core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -1183,7 +1183,7 @@ protected void prepareCH() {
      * For landmarks it is required to always call this method: either it creates the landmark data or it loads it.
      */
     protected void loadOrPrepareLM() {
-        boolean tmpPrepare = lmFactoryDecorator.isEnabled();
+        boolean tmpPrepare = lmFactoryDecorator.isEnabled() && !lmFactoryDecorator.getPreparations().isEmpty();
```

Comment:
Not sure if the isEmpty is necessary, but we had that check in the GraphHopperModule before, so I thought it would make sense to add it here as well.